### PR TITLE
Specify PEP 621

### PR DESCRIPTION
**Pull request by brettcannon**

### CODE REVIEW

File: source/specifications/declaring-project-metadata.rst [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+  later.
+- Build back-ends MUST raise an error if the metadata specifies a
+  field statically as well as being listed in ``dynamic``.
+- If the metadata does not list a field in ``dynamic``, then a build
+  back-end CANNOT fill in the requisite metadata on behalf of the user
+  (i.e. ``dynamic`` is the only way to allow a tool to fill in
+  metadata and the user must opt into the filling in).
+- Build back-ends MUST raise an error if the metadata specifies a
+  field in dynamic but the build back-end was unable to provide the
+  data for it.
```
Comment (pfmoore) [inline source/specifications/declaring-project-metadata.rst:295]: Maybe clarify this? Backends may determine dynamically that the correct value is "omitted", and therefore the final value won't be present, without that being an error. Maybe just add a second sentence to that effect?

File: source/specifications/declaring-project-metadata.rst [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+
+==========================
+Declaring project metadata
+==========================
+
+:pep:`621` specifies how to write a project's
+:ref:`core metadata <core-metadata>` in a ``pyproject.toml`` file for
+packaging-related tools to consume. It defines the following
+specification as the canonical source for the format used.
+
```
Comment (pfmoore) [inline source/specifications/declaring-project-metadata.rst:11]: There should be a brief description here of what "static" and "dynamic" mean, before you use the terms in the next paragraph.

**Review by pfmoore — CHANGES_REQUESTED** [review]

**Comment by brettcannon** [comment]
@pfmoore PTAL

**Review by pfmoore — APPROVED** [review]

**Comment by brettcannon** [comment]
@pfmoore do you want me to go ahead and merge this, or wait until PEP 621 is no longer provisional?

**Comment by pfmoore** [comment]
I think it should be merged now.

**Pull request merged**